tests/nvmeof/test_ceph_nvmeof_alerts_events.py

```diff
@@ -0,0 +1,156 @@
+from tests.rbd.rbd_utils import initial_rbd_config
+from utility.log import Log
+from utility.retry import retry
+
+LOG = Log(__name__)
+
+
+def test_ceph_83610948(ceph_cluster, config):
+    """[CEPH-83610948] Raise a healthcheck when a gateway is in UNAVAILABLE state.
+
+    This test case validates the warning which would be get hit on any gateway unavailability
+    and Raise a healthcheck warning under ceph status and ceph health.
+
+    Args:
+        ceph_cluster: Ceph cluster object
+        config: test case config
+    """
+    rbd_pool = config["rbd_pool"]
+    deploy_nvme_service(ceph_cluster, config)
+    ha = HighAvailability(ceph_cluster, config["gw_nodes"], **config)
+
+    with parallel() as p:
+        for subsys_args in config["subsystems"]:
+            subsys_args["ceph_cluster"] = ceph_cluster
+            p.spawn(configure_subsystems, rbd_pool, ha, subsys_args)
+
+    # Validate the Health warning states
+    @retry(ValueError, tries=6, delay=5)
+    def check_warning_status(gw, alert=True):
+        health, _ = ha.orch.shell(args=["ceph", "health", "detail", "--format", "json"])
+        health = loads(health)
+
+        if not alert:
+            if "NVMEOF_GATEWAY_DOWN" not in health["checks"]:
+                LOG.info(
+                    f"NVMEOF_GATEWAY_DOWN health check warning alerted disappeared - {health}."
+                )
+                if ha.check_gateway_availability(gw.ana_group_id, state="AVAILABLE"):
+                    return True
+            raise ValueError(f"NVMEOF_GATEWAY_DOWN warning still appears!!! - {health}")
+
+        if alert and "NVMEOF_GATEWAY_DOWN" in health["checks"]:
+            nvme_gw_down = health["checks"]["NVMEOF_GATEWAY_DOWN"]
+            if HEALTH_CHECK_WARN == nvme_gw_down:
+                LOG.info(
+                    f"NVMEOF_GATEWAY_DOWN health check warning alerted SUCCESSFULLY - {nvme_gw_down}."
+                )
+                if ha.check_gateway_availability(gw.ana_group_id, state="UNAVAILABLE"):
+                    return True
+            raise ValueError(
+                f"NVMEOF_GATEWAY_DOWN is alerted, but different messages - {nvme_gw_down}"
+            )
+        raise ValueError(f"NVMEOF_GATEWAY_DOWN health warning not alerted - {health}.")
+
+    # Fail a Gateway
+    fail_gw = choice(ha.gateways)
+    HEALTH_CHECK_WARN = {
+        "severity": "HEALTH_WARN",
+        "summary": {
+            "message": "1 gateway(s) are in unavailable state; gateway might be down, try to redeploy.",
+            "count": 1,
+        },
+        "detail": [
+            {
+                "message": f"NVMeoF Gateway 'client.{fail_gw.daemon_name}' is unavailable."
+            },
+        ],
+        "muted": False,
+    }
+    if not ha.system_control(fail_gw, action="stop", wait_for_active_state=False):
+        raise Exception(f"[ {fail_gw.daemon_name} ] GW Stop failed..")
+    check_warning_status(fail_gw)
+
+    # Start GW back and alert should not be visible
```

Comment:
Why would alert not be visible once GW comes up? lets discuss on validation .

Reply:
if GW is active, alert is not required. I dont see the reason why we need alert for ACTIVE gateway.

Reply:
We are actually listing GWs and GW groups in ceph -s from 8.1 active/inactive

Reply:
@rahullepakshi I did not get your comment. 
But on the GW DOWN alert we have two validations, 
1. Validate the warning message from `ceph health detail`.
2. Another one to check GW availability inside `check_warning_status()`

Reply:
@sunilkumarn417 Please ignore that. this looks good. I can explain better once I have 8.1 cluster